### PR TITLE
Fix broken build on Windows/MinGW

### PR DESCRIPTION
**Pull request by moodmosaic**
This was removed in https://github.com/hedgehogqa/haskell-hedgehog/pull/271/commits/43d7e272b4f46f11a177f494cb6202066bbca556#diff-724bc13f92f28d11e3df4525f10ed86a but it shouldn't.

### CODE REVIEW

**Comment by moodmosaic** [comment]
Before this PR, error message on Windows is:

```
    C:\Snapshot\src\haskell-hedgehog\hedgehog\src\Hedgehog\Internal\Runner.hs:299:5: error:
        Variable not in scope: hSetEncoding :: t0 -> t1 -> IO a0
        |
    299 |     hSetEncoding stdout utf8
        |     ^^^^^^^^^^^^

    C:\Snapshot\src\haskell-hedgehog\hedgehog\src\Hedgehog\Internal\Runner.hs:299:18: error:
        Variable not in scope: stdout
        |
    299 |     hSetEncoding stdout utf8
        |                  ^^^^^^

    C:\Snapshot\src\haskell-hedgehog\hedgehog\src\Hedgehog\Internal\Runner.hs:299:25: error:
        * Variable not in scope: utf8
        * Perhaps you want to add `utf8' to the import list
          in the import of `Hedgehog.Internal.Gen'
          (src\Hedgehog\Internal\Runner.hs:34:1-66).
        |
    299 |     hSetEncoding stdout utf8
        |                         ^^^^

    C:\Snapshot\src\haskell-hedgehog\hedgehog\src\Hedgehog\Internal\Runner.hs:300:5: error:
        Variable not in scope: hSetEncoding :: t2 -> t3 -> IO a1
        |
    300 |     hSetEncoding stderr utf8
        |     ^^^^^^^^^^^^

    C:\Snapshot\src\haskell-hedgehog\hedgehog\src\Hedgehog\Internal\Runner.hs:300:18: error:
        Variable not in scope: stderr
        |
    300 |     hSetEncoding stderr utf8
        |                  ^^^^^^

    C:\Snapshot\src\haskell-hedgehog\hedgehog\src\Hedgehog\Internal\Runner.hs:300:25: error:
        * Variable not in scope: utf8
        * Perhaps you want to add `utf8' to the import list
          in the import of `Hedgehog.Internal.Gen'
          (src\Hedgehog\Internal\Runner.hs:34:1-66).
        |
    300 |     hSetEncoding stderr utf8
        |                         ^^^^

Nikos@WIN-N4O1DOA8L9V MINGW64 ~/src/haskell-hedgehog (master)
```

**Comment by jystic** [comment]
Damn I thought it was only needed for `runTests`

**Comment by jystic** [comment]
Maybe we should add https://www.appveyor.com/ ?

**Comment by moodmosaic** [comment]
Yeah or Azure Pipelines (the Azure CI). Having this just merged for now should be good enough I guess.

**Comment by jystic** [comment]
I have appveyor on one of my other packages so i just copied the yml more or less, happy to use Azure though if you set it up?

Lol looks like appveyor only gives 1 build bot, maybe I shouldn't have added so many ghc versions.

**Comment by moodmosaic** [comment]
That was fast! Let's see what AppVeyor says now.

**Comment by jystic** [comment]
``'${CABAL}' is not recognized as an internal or external command,```
Caught out by windows vs unix env vars 😆

**Comment by moodmosaic** [comment]
Yep, I can take a look at Azure pipelines and see.

**Comment by jystic** [comment]
I'm not sure if this bit is going to work, I don't know what would cause those directories to be created or if parens `()` even work in the windows shell:
```
 # cabal check
 - "(cd hedgehog-* && %CABAL% check)"
 - "(cd hedgehog-dieharder-* && %CABAL% check)"
 - "(cd hedgehog-example-* && %CABAL% check)"
 - "(cd hedgehog-quickcheck-* && %CABAL% check)"
 - "(cd hedgehog-test-laws-* && %CABAL% check)"
```

**Comment by jystic** [comment]
Hooray for a windows package manager, not like the bad old days :)

**Comment by moodmosaic** [comment]
As a last resort we can use Stack for the Windows build and run have it run `stack test`.

**Comment by jystic** [comment]
Oh i'm sure the tests will work, I don't even know what `check` does, just copied it from `.travis.yml`

**Comment by moodmosaic** [comment]
Yeah I'm not sure `choco install ghc cabal` will work but let's see, I could be wrong.

File: .appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    - GHC_VERSION: "7.10.2"
+
+cache:
+ - "C:\\SR"
+
+install:
+ - "choco install -y ghc --version %GHC_VERSION%"
+ - "choco install -y cabal --version %CABAL_VERSION%"
+ - "refreshenv"
+ - "set PATH=C:\\msys64\\mingw64\\bin;C:\\msys64\\usr\\bin;%PATH%"
```
Comment (moodmosaic) [inline .appveyor.yml]: Not sure if we need to set those mingw64 env variables since GHC on Windows uses its own mingw64 binaries.

Comment (jystic) [reply .appveyor.yml]: I'll try remove them

Comment (jystic) [reply .appveyor.yml]: Maybe it was so we can use posix shell commands in the script?

Comment (moodmosaic) [reply .appveyor.yml]: Maybe yes

Comment (jystic) [reply .appveyor.yml]: I've removed it, let's see what happens

File: .appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+environment:
+  global:
+    CABAL: "cabal --store-dir=C:\\SR --http-transport=plain-http"
+    CABAL_VERSION: "2.4.1.0"
+    HC: "ghc"
+    TEST: "--enable-tests"
+    NOTEST: "--disable-tests"
+    BENCH: "--enable-benchmarks"
+    NOBENCH: "--enable-benchmarks"
```
Comment (moodmosaic) [inline .appveyor.yml]: Shouldn't this be `--disable-benchmarks` instead?

**Pull request merged**